Supervision Work/src/uk/ac/cam/ap886/oopjava/supervisionProjects/Hangman/ArrayDictionary.java

```diff
@@ -0,0 +1,50 @@
+package uk.ac.cam.ap886.oopjava.supervisionProjects.Hangman;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Random;
+
+public class ArrayDictionary implements Dictionary {
+	private ArrayList <String> words=new ArrayList<String>();
+
+	//------------------------------------------
+	public ArrayDictionary(String filename){
```

Comment:
Not convinced that loading the file should be the responsibility of a Dictionary. A Dictionary is strictly a datastructure. If someone gave me an "ArrayDictionary" I would not expect it to be loading anything itself. Dependency injection, giving it some sort of "Loader" interface, would make more sense.

Also, stylistic (and very subjective thing) but I personally think that constructors shouldn't explicitly do any work. Personally. Really doesn't matter too much though.